client/src/pages/Login.js

```diff
@@ -0,0 +1,75 @@
+import React, {Fragment, useState} from "react";
+import {useHistory} from 'react-router-dom';
+import {useAuth} from "../providers/auth/auth.provider";
+import {setIsAuthenticated, fetchUserSuccess, fetchUserRequest, fetchUserFailure} from "../providers/auth/auth.action";
+
+import {AuthForm, RedirectDiv} from './Signup'
+import axios from "axios";
+import {Grid} from "@material-ui/core";
+
+const Login = () => {
+    const auth = useAuth();
+    const {dispatchIsAuthenticated, dispatchUser} = auth;
+    const [serverResponse, setServerResponse] = useState('');
+
+    const history = useHistory();
+
+    //Callback for the form submission after validating successfully
+    const onSubmit = (data) => {
+
+        const {email, password} = data;
+
+        setServerResponse('');
+
+        //Make request to check whether the email and password are valid
+        const url = '/api/v1/auth/login'; //It is mock data, it will change when sever can provide auth api
+
+        dispatchUser(fetchUserRequest());
+
+        axios.post(url, {email, password})
+            .then(res => {
+                //If success to create a new account, redirect to login page
+                if (res.data.success){
+                    //Save data on local storage
+                    localStorage.setItem('isAuthenticated', true);
+                    localStorage.setItem('user', JSON.stringify(res.data.user));
+
+                    //Update the state of Auth providers
+                    dispatchIsAuthenticated(setIsAuthenticated(true));
+                    dispatchUser(fetchUserSuccess(res.data.user))
+                    console.log('login successfully');
+
+                    //Redirect to dashboard
+                    history.push('/');
+
+                } else {
+                    setServerResponse(res.data.error);
+                }
+
+            })
+            .catch(error => {
+                dispatchUser(fetchUserFailure(error.response.data.user))
+                setServerResponse(error.response.data.error);
+                console.log(error.response.data.error);
```

Comment:
Let's use async/await on the frontend, it makes the code more readable

Reply:
onSubmit  is a callback that will be call be the click event. I can change it to async/await function, but I do not know where I could put the error handle on the click event